### PR TITLE
Add Mesos-CNI documentation

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<!--- master only -->
+> ![warning](../images/warning.png) This document applies to the HEAD of the calico-containers source tree.
+>
+> View the calico-containers documentation for the latest release [here](https://github.com/projectcalico/calico-containers/blob/v0.19.0/README.md).
+<!--- else
+> You are viewing the calico-containers documentation for release **release**.
+<!--- end of master only -->
+
+# Calico-CNI with the Unified Containerizer - Manual Install Guide
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: Do we know what the right term is here? Unified containerizer vs Mesos containerizer?

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This guide explains how to add Calico networking to a Mesos Agent using CNI.
+Specifically, we will add the Calico plugin to perform networking
+for the **Unified Containerizer**.
+- If you're looking for information on installing Calico with the Docker Containerizer, see [Docker Containerizer Manual Install Guide](./ManualInstallCalicoDockerContainerizer.md)
+- If you're not sure the difference between the Unified and Docker Containerizers, see  [Mesos' information on Containerizers](http://mesos.apache.org/documentation/latest/containerizer/) and [Our Readme on Calico's integration for each](./README.md).
+
+## Prerequisites
+- **Mesos Cluster with v0.29.0+:** You must have an [agent with CNI enabled](https://github.com/apache/mesos/blob/master/docs/cni.md#configuring-cni-networks) that is apart of a running Mesos Cluster.
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: typo: apart -> "a part"

I think in general I'd reword this:

"You're Mesos agents must have CNI enabled" (I think it's obvious that Mesos should be running)

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- If you're not sure the difference between the Unified and Docker Containerizers, see  [Mesos' information on Containerizers](http://mesos.apache.org/documentation/latest/containerizer/) and [Our Readme on Calico's integration for each](./README.md).
+
+## Prerequisites
+- **Mesos Cluster with v0.29.0+:** You must have an [agent with CNI enabled](https://github.com/apache/mesos/blob/master/docs/cni.md#configuring-cni-networks) that is apart of a running Mesos Cluster. 
+    > Note: During configuration, you will have specified a `network_cni_config_dir` and `network_cni_plugins_dir`. We'll refer to these going forward as `$NETWORK_CNI_CONFIG_DIR` and `$NETWORK_CNI_PLUGINS_DIR`, respectively.
+- **Docker:** Calico's core services are run from within a Docker container on each agent, so you'll need to follow the relevant [docker installation guide](https://docs.docker.com/v1.8/installation/) on each.
+- **Etcd:** Calico uses etcd as its datastore. See the [Cluser Preparation guide](MesosClusterPreparation.md#etcd) for information on how to quickly get one running.
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: The linked preparation guide assumes you're running on CentOS.  Could we just link to the etcd documentation instead or in addition?

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## Install Calico
+1. Download Calico's CNI plugin:
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: nit: "Calico's CNI plugin" -> "The Calico CNI plugin"

To be consistent with how we refer to it elsewhere.

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    ```
+    curl -L -o $NETWORK_CNI_PLUGINS_DIR/calico \
+    https://github.com/projectcalico/calico-cni/releases/download/v1.3.0/calico
+    chmod +x $NETWORK_CNI_PLUGINS_DIR/calico
+    ```
+
+2. Run Calico Node, a Docker container with calico's core routing processes.
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: nit: Run `calico/node`

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+- If you're not sure the difference between the Unified and Docker Containerizers, see  [Mesos' information on Containerizers](http://mesos.apache.org/documentation/latest/containerizer/) and [Our Readme on Calico's integration for each](./README.md).
+
+## Prerequisites
+- **Mesos Cluster with v0.29.0+:** You must have an [agent with CNI enabled](https://github.com/apache/mesos/blob/master/docs/cni.md#configuring-cni-networks) that is apart of a running Mesos Cluster. 
+    > Note: During configuration, you will have specified a `network_cni_config_dir` and `network_cni_plugins_dir`. We'll refer to these going forward as `$NETWORK_CNI_CONFIG_DIR` and `$NETWORK_CNI_PLUGINS_DIR`, respectively.
+- **Docker:** Calico's core services are run from within a Docker container on each agent, so you'll need to follow the relevant [docker installation guide](https://docs.docker.com/v1.8/installation/) on each.
+- **Etcd:** Calico uses etcd as its datastore. See the [Cluser Preparation guide](MesosClusterPreparation.md#etcd) for information on how to quickly get one running.
+
+## Install Calico
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: Where does one run these steps? Every agent? On the master?

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+2. Run Calico Node, a Docker container with calico's core routing processes. 
+  The `calico-node` container can easily be launched using
+  `calicoctl`, Calico's command line tool. When doing so,
+  we must provide the location of the running etcd instance
+  by setting the `ECTD_AUTHORITY` environment variable.
+    ```
+    curl -L -o ./calicoctl https://github.com/projectcalico/calico-containers/releases/download/v0.18.0/calicoctl
+    chmod +x calicoctl
+    sudo ETCD_AUTHORITY=<etcd-ip:port> ./calicoctl node
+    ```
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: You should have more at the end here, even if it is just a "Calico should now be installed, for information on how to use it see our usage guide".

Basically, someone following this guide won't know what to do next when they get here.

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+# Calico-CNI with the Unified Containerizer - Manual Install Guide
+This guide explains how to add Calico networking to a Mesos Agent using CNI.
+Specifically, we will add the Calico plugin to perform networking
+for the **Unified Containerizer**.
+- If you're looking for information on installing Calico with the Docker Containerizer, see [Docker Containerizer Manual Install Guide](./ManualInstallCalicoDockerContainerizer.md)
+- If you're not sure the difference between the Unified and Docker Containerizers, see  [Mesos' information on Containerizers](http://mesos.apache.org/documentation/latest/containerizer/) and [Our Readme on Calico's integration for each](./README.md).
+
+## Prerequisites
+- **Mesos Cluster with v0.29.0+:** You must have an [agent with CNI enabled](https://github.com/apache/mesos/blob/master/docs/cni.md#configuring-cni-networks) that is apart of a running Mesos Cluster. 
+    > Note: During configuration, you will have specified a `network_cni_config_dir` and `network_cni_plugins_dir`. We'll refer to these going forward as `$NETWORK_CNI_CONFIG_DIR` and `$NETWORK_CNI_PLUGINS_DIR`, respectively.
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: "During configuration"

during what configuration?  Is this a "when starting the mesos agent" instead?

File: docs/mesos/UsageGuideCNI.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+- zookeeper
+- Mesos-Slave v0.29.0+ with the following true for each agent:
+    - Docker installed
+    - calico-node running
+    - calico-cni binary installed
+    - calico-ipam binary installed
+
+If your Slave does not meet these specifications simply follow the [CNI Manual Installation guide for Mesos](ManualInstallCalicoCNI.md) before continuing.
+
+## Configuring a Calico CNI Network
```
Comment (caseydavenport) [inline docs/mesos/UsageGuideCNI.md]: I think this should be a part of the guide above - this is really an "installing Calico" type step.

File: docs/mesos/UsageGuideCNI.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+This assumes a running Mesos cluster that meets the following specifications:
+
+- Mesos-Master v0.29.0+
+- etcd
+- zookeeper
+- Mesos-Slave v0.29.0+ with the following true for each agent:
+    - Docker installed
+    - calico-node running
+    - calico-cni binary installed
+    - calico-ipam binary installed
```
Comment (caseydavenport) [inline docs/mesos/UsageGuideCNI.md]: They don't actually need the calico-ipam binary installed right now.  It ships as part of the `calico` binary.

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+- If you're not sure the difference between the Unified and Docker Containerizers, see  [Mesos' information on Containerizers](http://mesos.apache.org/documentation/latest/containerizer/) and [Our Readme on Calico's integration for each](./README.md).
+
+## Prerequisites
+- **A Running Mesos v0.29.0+ cluster**  [with CNI enabled on each agent](https://github.com/apache/mesos/blob/master/docs/cni.md#configuring-cni-networks).
+
+> Note: When enabling CNI in Mesos, you will have specified a `network_cni_config_dir` and `network_cni_plugins_dir`. We'll refaer to these going forward as `$NETWORK_CNI_CONFIG_DIR` and `$NETWORK_CNI_PLUGINS_DIR`, respectively.
+
+- **Docker** must be installed and running on each agent in order to run `calico/node`. Follow the relevant [docker installation guide](https://docs.docker.com/engine/installation/).
+- **Etcd** is used by Calico to store network configurations. See [etcd's docker standalone guide](https://coreos.com/etcd/docs/latest/docker_guide.html) for information on how to quickly get an instance running.
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: Lower case "e" in `etcd`

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+        "type": "calico-ipam"
+    },
+    "etcd_authority": "<etcd-ip:port>:2379"
+}
+```
+
+Mesos will actively scan that directory when launching containers, so without needing to restart the Agent process, you are now ready to launch calico-CNI Mesos tasks.
+
+## Launching Tasks
+With our network configured, we can launch tasks using `mesos-execute`:
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: I'd just add one or two more lines to say something like "we tell mesos-execute to use the Calico CNI plugin by specifying the name of the network we created in the previous step" or something like that.

Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: Is there an issue for supporting this is Marathon? Might want to also say something like "Marathon support will be coming soon, please see this issue".  Not sure.

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+Run the following steps on each agent.
+
+1. Download the Calico CNI plugin:
+    ```
+    curl -L -o $NETWORK_CNI_PLUGINS_DIR/calico \
+        https://github.com/projectcalico/calico-cni/releases/download/v1.3.0/calico
+    chmod +x $NETWORK_CNI_PLUGINS_DIR/calico
+    ```
+
+2. Run `calico/node`, a Docker container with calico's core routing processes.
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: `calico/node` doesn't need to be run on the master?

File: docs/mesos/ManualInstallCalicoCNI.md (hunk 12 above)
Comment (djosborne) [inline docs/mesos/ManualInstallCalicoCNI.md]: Got the first point. But no Marathon issue open yet since a stable mesos release with this functionality hasn't even been released yet :(

File: docs/mesos/ManualInstallCalicoCNI.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+## Launching Tasks
+With our network configured, we can use `mesos-execute` to launch a task on the CNI network we just configured: 
+```
+mesos-execute --containerizer=mesos \
+              --name=cni \
+              --master=172.17.0.4:5050 \
+              --networks=my-net-1 \
+              --command=ifconfig
+```
+This will launch a simple task that prints its IP address and exits. Since the task is launched on a calico network, it will report an IP from the default Calico pool: `192.16.0.0/16`
```
Comment (caseydavenport) [inline docs/mesos/ManualInstallCalicoCNI.md]: If you're going to talk about pools, you should probably link to information on how to configure them.

This still doesn't explicitly call out the `--networks` flag.  I think you want something like:

"Pay attention to the `--networks=my-net-1` option.  This tells Mesos to launch the task using the CNI network we defined above"

**Comment by caseydavenport** [comment]
One more comment, then LGTM.
